src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -108,20 +109,30 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
         [HttpPost]
         [Route("admin/host/drain")]
         [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
-        public IActionResult Drain([FromServices] IScriptHostManager scriptHostManager)
+        public async Task<IActionResult> Drain([FromServices] IScriptHostManager scriptHostManager)
         {
             _logger.LogDebug("Received request for draining host");
 
-            if (Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager))
-            {
-                // Stop call to some listeners get stuck, Not waiting for the stop call to complete
-                drainModeManager.EnableDrainModeAsync(CancellationToken.None).ConfigureAwait(false);
-                return Ok();
-            }
-            else
+            if (!Utility.TryGetHostService(scriptHostManager, out IDrainModeManager drainModeManager))
             {
                 return StatusCode(StatusCodes.Status503ServiceUnavailable);
             }
+
+            await _drainModeSemaphore.WaitAsync();
+
+            // Stop call to some listeners gets stuck, not waiting for the stop call to complete
+            _ = drainModeManager.EnableDrainModeAsync(CancellationToken.None)
+                                .ContinueWith(
+                                    antecedent =>
+                                    {
+                                        if (antecedent.Status == TaskStatus.Faulted)
+                                        {
+                                            _logger.LogError("Something went wrong enabling drain mode", antecedent.Exception);
```

Comment:
This is subtle -- but I think the exception needs to go first here? This works b/c there's an overload that it'll match, but it won't recognize that second param as an extension:

```
// You want this one...
LogError(ILogger, Exception, String, Object[])

// But what you have will resolve to this one...
LogError(ILogger, String, Object[])
```